Core/src/IfcGeometryConverter/FaceConverter.h

```diff
@@ -1032,7 +1032,9 @@ namespace OpenInfraPlatform {
 				*/
 
 				std::vector<std::vector<carve::geom::vector<3>>> convertIfcCartesianPoint2DVector(
-					const std::vector<std::vector<EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>>>& points2D)  const throw(...)
+					const OpenInfraPlatform::EarlyBinding::EXPRESSContainer<
+					OpenInfraPlatform::EarlyBinding::EXPRESSContainer<
+					EXPRESSReference<typename IfcEntityTypesT::IfcCartesianPoint>, 0, -1>, 0, -1>& points2D)  const throw(...)

```

Comment:
Currently, it works. However, the long typename
`OpenInfraPlatform::EarlyBinding::EXPRESSContainer<OpenInfraPlatform::EarlyBinding::EXPRESSContainer<EXPRESSRefrence<...`
looks a bit odd.
A simple reduction to `EXPRESSContainer<EXPRESSContainer<EXPRESSReference<...` does not compile because of "missing type specifier".
Does someone has a suggestion how to improve this?

Reply:
All `EXPRESS...` classes are added to the `oip` namespace, so sth like `oip::EXPRESSContainer` should work.

Reply:
Currently, `oip::EXPRESSContainer` is not available in the `oip` namespace:
![210515 oip_EXPRESS__](https://user-images.githubusercontent.com/64251870/118404753-7d533a00-b674-11eb-896a-b4f0b7bd210f.png)

I found three possible solutions:
- Using of the existing type alias `using LIST` from [EXPRESSContainer.h](https://github.com/tumcms/Open-Infra-Platform/blob/76cd098fdc663ddf1983f214e559f1b92d91a0f0/EarlyBinding/src/EXPRESS/EXPRESSContainer.h#L233). The variable declaration in the `FaceConverter` would look like
`oip::LIST<0, -1, oip::LIST<0, -1, EXPRESSReference<...>>>`
This solution is not very nice and intuitive because the `EXPRESSContainer` is hidden behind the alias `LIST`.
- Including `EXPRESSContainer` to the namespace `oip` by adding line 245 in EXPRESSContainer.h:
![210516 oip_EXPRESSContainer into namespace](https://user-images.githubusercontent.com/64251870/118406047-feadcb00-b67a-11eb-88d0-33bcc1c4a47e.png)
The variable declaration in the `FaceConverter` would look like
`oip::EXPRESSContainer<oip::EXPRESSContainer<EXPRESSReference<...>, 0, -1>, 0, -1>`
This is similar to the way how all other `EXPRESS...` classes are handled.
- Adding of a new type alias in line 240 of EXPRESSContainer.h
![210516 oip_EXPRESSContainer using alias](https://user-images.githubusercontent.com/64251870/118406618-d58e3a00-b67c-11eb-8ffc-bca8c922e466.png)
The variable declaration in the `FaceConverter` would look like
`EXPRESSContainer<EXPRESSContainer<EXPRESSReference<...>>>`
This is the way how the `EXPRESSReference` type is used almost everywhere. Probably, it could be the simplest and most intuitive declaration for instances of `EXPRESSContainer`.

I want to ask for some advice to this suggestions because of the possible big side effects of changes in the `EXPRESS...` classes (which was generally mentioned in a previous dev-meeting).

Reply:
For the time being, use option 3 but add the typename definition in the FaceConverter.h file (NOT EXPRESSContainer). We need to discuss this further, but this should allow you to continue programing.

Also: use (similarly to the definition of the `ControlPointsList`) 0 and LIST_MAXSIZE as the integer template parameters.

Reply:
Done in commit 48612b2